encrypted-media-respec.html

```diff
@@ -2331,15 +2365,16 @@
 
         <section id="queue-waitingforkey">
           <h4>Queue a "waitingforkey" Event</h4>
-          <p>The Queue a "waitingforkey" Event algorithm is run when the CDM cannot decrypt the current block.
+          <p>The Queue a "waitingforkey" Event algorithm is run when the CDM cannot decrypt an encrypted block.
             It should only be called when the <a def-id="htmlmediaelement"></a> object is <a def-id="potentially-playing"></a> and its <a def-id="readystate"></a> is equal to <a def-id="have-future-data"></a> or greater.  
```

Comment:
Firefox won't advance past HAVE_CURRENT_DATA until we have successfully decoded the data at the current playback position, i.e. until we've had usable keys and used them. So we wouldn't be able to fire this event... uh... ever, because not having the key for the current playback position means our readyState we'll drop back below HAVE_FUTURE_DATA.

Can we remove the HAVE_FUTURE_DATA check? What is the readyState check trying to achieve that the potentially playing check and the checks before calling this algorithm don't achieve?